crates/fluvio-cluster/src/start/local.rs

```diff
@@ -373,29 +381,56 @@ impl LocalInstaller {
     /// Checks if all of the prerequisites for installing Fluvio locally are met
     /// and tries to auto-fix the issues observed
     pub async fn setup(&self) -> CheckResults {
-        println!("Performing pre-flight checks");
+        let mut receiver = self.setup_with_progress().await;
+        let mut check_results = CheckResults::default();
+
+        while let Some(local_progress) = receiver.next().await {
+            if let LocalSetupProgressMessage::Check(check_result) = local_progress {
+                check_results.push(check_result);
+            }
+        }
+        check_results
+    }
+
+    /// Checks if all of the prerequisites for installing Fluvio locally are met
+    /// and tries to auto-fix the issues observed
+    pub async fn setup_with_progress(&self) -> Receiver<LocalSetupProgressMessage> {
+        let (sender, receiver) = futures_channel::mpsc::channel(100);
+        fluvio_future::task::spawn(self.clone().setup_with_progress_impl(sender));
+        receiver
+    }
+
+    async fn setup_with_progress_impl(
+        self,
+        mut sender: Sender<LocalSetupProgressMessage>,
+    ) -> Result<(), ClusterError> {
         let mut sys_config: ChartConfig = ChartConfig::sys_builder()
             .version(self.config.chart_version.clone())
             .build()
             .expect("should build config since all required arguments are given");
 
         if let Some(location) = &self.config.chart_location {
-            sys_config.location = location.to_owned().into();
+            sys_config.location = location.clone().into();
         }
 
-        if self.config.render_checks {
-            let mut progress = ClusterChecker::empty()
-                .with_local_checks()
-                .with_check(SysChartCheck::new(sys_config))
-                .run_and_fix_with_progress();
-            render_check_progress(&mut progress).await
-        } else {
-            ClusterChecker::empty()
-                .with_local_checks()
-                .with_check(SysChartCheck::new(sys_config))
-                .run_wait_and_fix()
-                .await
-        }
+        let cluster_checker = ClusterChecker::empty()
+            .with_local_checks()
+            .with_check(SysChartCheck::new(sys_config));
+
+        let checks_number = cluster_checker.checks_number() as u64;
+        sender
+            .send(LocalSetupProgressMessage::PreFlighCheck(checks_number))
+            .await
+            .map_err(LocalInstallError::ChannelSendError)?;
+
+        let mut progress = cluster_checker
+            .run_and_fix_with_progress()
+            .map(|check| Ok(LocalSetupProgressMessage::Check(check)));
+        sender
+            .send_all(&mut progress)
+            .await
+            .map_err(LocalInstallError::ChannelSendError)?;
```

Comment:
The key difference is here. `progess` is a stream and `sender` is a sink, so we can plug them together with `send_all` instead of needing to loop to push events through